nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java

```diff
@@ -136,45 +117,38 @@ public void onTrigger(ProcessContext context, ProcessSessionFactory sessionFacto
         // create a session and try to get a FlowFile, if none available then close any idle senders
         final ProcessSession session = sessionFactory.createSession();
         final FlowFile flowFile = session.get();
+
         if (flowFile == null) {
-            final PruneResult result = pruneIdleSenders(context.getProperty(IDLE_EXPIRATION).asTimePeriod(TimeUnit.MILLISECONDS).longValue());
-            // yield if we closed an idle connection, or if there were no connections in the first place
-            if (result.getNumClosed() > 0 || (result.getNumClosed() == 0 && result.getNumConsidered() == 0)) {
-                context.yield();
-            }
             return;
         }
 
-        // get a sender from the pool, or create a new one if the pool is empty
-        // if we can't create a new connection then route flow files to failure and yield
-        // acquireSender will handle the routing to failure and yielding
-        ChannelSender sender = acquireSender(context, session, flowFile);
-        if (sender == null) {
+        if (eventSender == null) {
             return;
         }
 
-        try {
-            String delimiter = context.getProperty(MESSAGE_DELIMITER).evaluateAttributeExpressions(flowFile).getValue();
-            if (delimiter != null) {
-                delimiter = delimiter.replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
-            }
+        String delimiter = context.getProperty(MESSAGE_DELIMITER).evaluateAttributeExpressions(flowFile).getValue();
+        if (delimiter != null) {
+            delimiter = delimiter.replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
+        }
 
-            // if no delimiter then treat the whole FlowFile as a single message
+        // if no delimiter then treat the whole FlowFile as a single message
+        try {
             if (delimiter == null) {
-                processSingleMessage(context, session, flowFile, sender);
+                processSingleMessage(context, session, flowFile, eventSender);
             } else {
-                processDelimitedMessages(context, session, flowFile, sender, delimiter);
+                processDelimitedMessages(context, session, flowFile, eventSender, delimiter);
             }
-
-        } finally {
-            relinquishSender(sender);
+        } catch (EventException e) {
+            session.transfer(flowFile, REL_FAILURE);
+            session.commitAsync();
+            context.yield();
         }
     }
 
     /**
      * Send the entire FlowFile as a single message.
      */
-    private void processSingleMessage(ProcessContext context, ProcessSession session, FlowFile flowFile, ChannelSender sender) {
+    private void processSingleMessage(final ProcessContext context, final ProcessSession session, final FlowFile flowFile, final EventSender<byte[]> sender) {
```

Comment:
Is it necessary to pass `EventSender` as parameter here since it is already a class member?